src/amo/reducers/versions.js

```diff
@@ -24,19 +26,25 @@ export type AddonVersionType = {
 export const createInternalVersion = (
   version: ExternalAddonVersionType,
 ): AddonVersionType => {
-  // TODO: We should find the correct file, for now we'll just take the first one.
-  // Note that multiple files per version is going away, but not quite yet.
-  const file = version.files[0];
-  return {
+  const addonVersion = {
     compatibility: version.compatibility,
-    created: file.created,
-    files: version.files,
-    filesize: file.size,
+    platformFiles: createPlatformFiles(version),
     id: version.id,
     license: { name: version.license.name, url: version.license.url },
     releaseNotes: version.release_notes,
     version: version.version,
   };
+
+  const file = findFileForPlatform({
+    platformFiles: addonVersion.platformFiles,
+    userAgentInfo: state.api.userAgentInfo, // this won't work currently
```

Comment:
My idea was that when we create the internal version, it would make sense to do this lookup and store the info we need from the correct `file` object. The issue (or one issue) is that I don't have access to `userAgentInfo` at this point.

Instead of doing this I could just leave `created` and `filesize` out of the `AddonVersionType` and calculate them on the fly, each time I need to render a version, but that seems like processing that's worth avoiding.

At this point I see a couple of options:

1. Do it like I've done it above, and pass `userAgentInfo` in the action payload which would result in the call to `createInternalVersion` (which will likely come from a saga).

2. Leave these properties out and create a function/functions that will return the `create` and `filesize`. I guess that function would still live in this reducer? I'd then have to call it inside a loop when I am rendering the versions.

@kumar303 what do you think?